tasks/test.py

```diff
@@ -1109,3 +1113,59 @@ def junit_macos_repack(_, infile, outfile):
     contain correct job name and job URL.
     """
     repack_macos_junit_tar(infile, outfile)
+
+
+@task
+def get_modified_packages(ctx) -> List[GoModule]:
+
+    modified_files = get_modified_files(ctx)
+    modified_go_files = [
+        f"./{file}" for file in modified_files if file.endswith(".go") or file.endswith(".mod") or file.endswith(".sum")
+    ]
+
+    modules_to_test = {}
+    go_mod_modified_modules = set()
+
+    for modified_file in modified_go_files:
+        match_precision = 0
+        best_module_path = None
+
+        # Since several modules can match the path we take only the most precise one
+        for module_path in DEFAULT_MODULES:
+            if module_path in modified_file:
+                if len(module_path) > match_precision:
```

Comment:
You are just counting the number of chars in the path, right? Should you measure the length of `module_path.split("/")` because `len("pkg/verylongmodulename") > len("pkg/short/name")`?

Reply:
I think this is fine to only count the length of the path string. Because if for example a package match several modules. Like for example a package in `internal/tools/proto` will match `internal/tools` and `internal/tools/proto` modules. But since the it can match several module only if one is included in the other (here `internal/tools/proto` path is included in `internal/tools`) so the most precise one contains the less precise one and so have more chars in the path. Let me know if it is not clear!

Reply:
Yes I was not sure, it's ok for me, thanks for precision